src/features/smSystem/suppliersOrders/components/AddOrderModal/Modal.tsx

```diff
@@ -0,0 +1,190 @@
+import {
+  Box,
+  Button,
+  CircularProgress,
+  Modal,
+  Stack,
+  TextField,
+  Typography,
+} from '@mui/material';
+import { GridRowParams } from '@mui/x-data-grid';
+import { SetStateAction, useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { Pages } from '../../../../../utils';
+import {BasicModalProps} from '../../../app/types/index';
+import { useCreateOrder } from '../../api/useCreateOrder';
+import { useGetSuppliers } from '../../api/useGetSuppliers';
+
+import BranchesTable from './BranchesTable';
+import SuppliersTable from './SuppliersTable';
+
+
+const AddOrderModal = ({ open, handleClose }: BasicModalProps) => {
+  const [page, setPage] = useState(0);
+  const [name, setName] = useState<string>('');
+  const [nameToSearch, setNameToSearch] = useState<string>('');
+  const [isCreating, setIsCreating] = useState<boolean>(false);
```

Comment:
```suggestion
  const [name, setName] = useState('');
  const [nameToSearch, setNameToSearch] = useState('');
  const [isCreating, setIsCreating] = useState(false);
```

you don't have to set type for states if the default value indicate the type, but in case when it's for example null or string, then you have for examp[le:
`const [name, setName] = useState<null | string>(null);`